docs/spec_draft.md

```diff
@@ -1917,6 +1918,70 @@ and produces a `result` tensor. More formally,
 ```
 
 [Back to Ops](#index-of-ops)
+
+## stablehlo.rng_bit_generator
+
+### Semantics
+
+Returns an `output` filled with uniform random bits and an updated output state
+`output_state` given an initial state `initial_state` using the pseudorandom
+number generator algorithm `rng_algorithm`. The output is guaranteed to be
+deterministic function of `initial_state`, but it is not guaranteed to be
+deterministic between backends and different compiler versions.
+
+`rng_algorithm` is one of the following, different variants of the algorithm is
+implementation-defined:
+  * `DEFAULT`: Backend specific algorithm.
+  * `THREE_FRY`: Threefry counter-based PRNG algorithm.*
+  * `PHILOX`: Philox algorithm to generate random numbers in parallel.*
+
+\* See: [Salmon et al. SC 2011. Parallel random numbers: as easy as 1, 2, 3.
+](http://www.thesalmons.org/john/random123/papers/random123sc11.pdf)
+
+### Inputs
+
+| Name            | Type                                         |
+|-----------------|----------------------------------------------|
+| `initial_state` | tensor of integer or floating-point type     |
+| `rng_algorithm` | enum of `DEFAULT`, `THREE_FRY`, and `PHILOX` |
+
+### Outputs
+
+| Name           | Type                                     |
+|----------------|------------------------------------------|
+| `output_state` | tensor of integer or floating-point type |
```

Comment:
As far as I can see from [lax.py](https://github.com/google/jax/blob/40e81c3a8655709c9aa554b52e1a85842786e034/jax/_src/lax/lax.py#L4373-L4380) and the investigation in this PR, both `initial_state` and `output_state` are "1-dimensional tensor of `ui64`"?

Reply:
`initial_state` and `output_state` for DEFAULT algorithm may be able to take n-dimensional tensor of element types other than `ui64`. Similar to my comment above, I propose that we restrict to 1-dimensional tensor of `ui64` and opening an issue to track this in the same ticket.

Reply:
Created issue #643

Reply:
Let's restrict to "1-dimensional tensor of type `u64`". From what I understand, it looks like the significant flexibility of `DEFAULT` might be useful privately to XLA, but the public interface is still `tensor<2xui64>`.